### PR TITLE
Issue/2910 batch add product tags

### LINKED ISSUE #2190: Edit Products M3 - Product tags Step 3: add new tags to a product
After [step 1](https://github.com/woocommerce/woocommerce-android/issues/2188) and [step 2](https://github.com/woocommerce/woocommerce-android/issues/2189), we should provide the option to add tags to a product.

Annotation on 2020-04-03 at 17-52-22

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #2190  by adding logic to add new tags to a product.

~**Note that this PR is in draft this till [bottom sheet PR](https://github.com/woocommerce/woocommerce-android/pull/2617) can be reviewed and merged.**~

### Changes
- When a tag is entered, the logic is to first check if this tag exists in the local db. (We are fetching 100 tags at a time so most of the tags should be available in the local db). 
  - If the tag exists in the local db, then the tag is added automatically to the product. 
  - If the tag does not exist in the local db, we can assume that the tag added is new, so we add it to a new list. When the user clicks on the "DONE" button, if this list is not empty, we first call an API to add these tags to the site. If the response is successful, the tag is automatically added to the product. If the response results in an error, a generic snackbar is displayed. 

#### Testing
- Click on a product from the Products list (without any tags).
- Click on the bottom sheet and click on `Tags`.
- Type a new tag in the text field. Notice the tag is displayed correctly and the "DONE" menu button is displayed.
- Click on the "DONE" menu button and notice that the add is added successfully to the site and that you are redirected to the Product detail screen.
- Notice the new tags are displayed in the Product detail screen.
- Click on the UPDATE button and verify that the tags are updated successfully for the product.
- Click on the Tags section for the same product.
- Remove the selected tags and click on DONE button.
- Notice the tags section is no longer visible. 
- Click on the UPDATE menu button.
- Verify that the tags are updated successfully for the product.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by anitaa1990** [comment]
@woocommerce/android-developers, This one is ready for review 👍

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/LiveDataExt.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.woocommerce.android.extensions
+
+import androidx.lifecycle.MutableLiveData
+
+fun <T> MutableLiveData<MutableList<T>>.addNewItem(item: T) {
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/LiveDataExt.kt:5]: Nice!

**Comment by anitaa1990** [comment]
Thanks for the review @0nko!!

> While testing, I noticed when you add an existing tag, the Done button doesn't appear so you can't save the changes.

🤔 I'm not able to reproduce this behaviour. I tried entering an existing tag in the EditText and also tried selecting a tag and in both cases the DONE button appears. Is there another scenario where you are facing this issue?


> Also, the Add more details button at the Product detail screen covers the bottom property 


Nice catch! Fixed in 33c0c78.

I also faced an error when trying to add an empty tag to the API. I fixed that in 733bfdc. Ready for another round 👍

**Comment by anitaa1990** [comment]
Hey @0nko, this one is ready for review 👍

**Comment by 0nko** [comment]
> I'm not able to reproduce this behaviour. 

Sorry, there seems to be a specific way, which I didn't realize: First add a new tag and then tap on an existing one:

tags

**Comment by 0nko** [comment]
@anitaa1990 Ah, after pulling the latest changes, the flow to reproduce the bug is a bit different:

1. Go to the Tags screen and add a new tag
2. Notice the Update button appears
3. Add an existing tag
4. Remove the last tag you added
5. Notice the Update button is gone

**Comment by anitaa1990** [comment]
Ahh, thanks for the additional details @0nko!! I was able to reproduce the. bug and fixed it in a1c8aaf. Ready for another round 👍

**Pull request merged**